vantage/virtual_tag_config_resource.go

```diff
@@ -0,0 +1,188 @@
+package vantage
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	tagsv2 "github.com/vantage-sh/vantage-go/vantagev2/vantage/virtual_tags"
+)
+
+type VirtualTagConfigResource struct {
+	client *Client
+}
+
+func NewVirtualTagConfigResource() resource.Resource {
+	return &VirtualTagConfigResource{}
+}
+
+func (r *VirtualTagConfigResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_virtual_tag_config"
+}
+
+func (r VirtualTagConfigResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
```

Comment:
curious why not use an autogen'd schema

Reply:
talked offline but in summary: we were working in parallel and I didn't explore that option.

I've added a follow-up task to the backlog to do that among some other things, like exploring relaxing resource requirements when we can infer an update operation